BookApi/BookApi/Repository/BookRepository.cs

```diff
@@ -0,0 +1,67 @@
+using BookApi.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace BookApi;
+
+public class BookRepository
+{
+    private BookContext bookContext;
+
+    public BookRepository(BookContext bookContext)
+    {
+        this.bookContext = bookContext;
+    }
+
+    public async Task<IList<Book>> GetAllBooks()
+    {
+        
+        return await bookContext.Books.ToListAsync();
+    }
+    
+    public async Task<Book> CreateBook(string title, int pageCount)
+    {
+        var book = new Book(title)
+        {
+            PageCount = pageCount
+        };
+
+        try
+        {
+            await this.bookContext.Books.AddAsync(book);
+            await bookContext.SaveChangesAsync();
+
+            return book;
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine(e);
```

Comment:
If you are only writing the error to the console and then throwing it again you can work without the try catch.